code/ai/aigoals.cpp

```diff
@@ -1042,13 +1088,12 @@ void ai_add_goal_sub_sexp( int sexp, ai_goal_type type, ai_info *aip, ai_goal *a
 		break;
 
 	case OP_AI_PLAY_DEAD:
-		aigp->priority = eval_num(CDR(node), priority_is_nan, priority_is_nan_forever);
-		aigp->ai_mode = AI_GOAL_PLAY_DEAD;
-		break;
-
+		// we don't clear out ship goals for the "persistent" goal variant
+		aigp->flags.set(AI::Goal_Flags::Clear_all_goals_first);
```

Comment:
IMHO wrapping this line in an `if`-check for `OP_AI_PLAY_DEAD` and putting it below the next `case` statement is easier to understand/maintain than using fall-through, but up to you.